### PR TITLE
Axios and cancellation

### PR DESCRIPTION
**Pull request by RikkiGibson**
Progress reporting is coming later. We have to make some decisions there because you actually do need to wrap the request stream to get upload progress in node.js, while in browser you need to pass a onDownload/UploadProgress handler to axios and get the raw xhr event (I expect we'll wrap this somehow?)

### CODE REVIEW

**Comment by RikkiGibson** [comment]
This passes all autorest.typescript tests

**Comment by RikkiGibson** [comment]
@daschult for some reason GitHub review UI is not cooperating but would like to get a review from you as well

File: test/shared/axiosHttpClientTests.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  });
+
+  it("should allow canceling requests", async function() {
+    // ensure that a large upload is actually cancelled
+    this.timeout(200);
+
+    const cancellationToken: CancellationTokenLike = {
+      setCancellationListener: (listener) => { cancellationToken.cancel = listener },
+      cancel: () => {}
+    };
```
[the remark this answers is not in the document]
Comment (RikkiGibson) [reply test/shared/axiosHttpClientTests.ts]: That's a good idea. Just to keep things straightforward I'd like to make it return `CancellationTokenLike`. Static method on ServiceClient?

`setCancellationListener` is the key part of the interface you provide to the HttpClient, so anyone trying to adapt AbortController or a vscode CancellationToken or similar will need to implement that method.

[the remark this answers is not in the document]
Comment (daschult) [reply test/shared/axiosHttpClientTests.ts]: I think I'd prefer it not to be a static method.
Oh, I just think I changed my mind. Why not add `createCancellationToken()` to the HttpClient interface? Cancellation is definitely a HTTP client feature, so the service client's `createCancellationToken()` method would just pass through to the stored HttpClient variable. Then all of the implementation specific details (the purpose of the `setCancellationListener()` method) can be hidden during creation time. Then the interface exposed to the customer would just be the `cancel()` method, right?

[the remark this answers is not in the document]
Comment (RikkiGibson) [reply test/shared/axiosHttpClientTests.ts]: We can, it just means that we have more of an obligation to provide a cancellation token type with all the bells and whistles on it (derived tokens, multiple listeners, etc). Which is maybe for the best. :)

[the remark this answers is not in the document]
Comment (RikkiGibson) [reply test/shared/axiosHttpClientTests.ts]: If we have users showing up with a different CancellationToken type that they want to use with our SDK, they can just add a listener to their own CancellationToken which cancels our CancellationToken. So I think we're all good.

File: lib/axiosHttpClient.ts [hunk 2]
```diff
@@ -101,13 +105,14 @@ export class AxiosHttpClient implements HttpClient {
         transformResponse: undefined,
         validateStatus: () => true,
         withCredentials: true,
+        maxContentLength: 1024*1024*1024*10,
```
Comment (daschult) [inline lib/axiosHttpClient.ts]: Is this defined in axios? Where does this number come from? I recommend a comment explanation here.

**Comment by RikkiGibson** [comment]
A little more work is needed to make abort available to users of the generated code but will merge now as this seems to be a pretty good milestone.

**Pull request merged**